check_diff/src/main.rs

```diff
@@ -15,11 +20,36 @@ struct CliInputs {
     rustfmt_config: Option<Vec<String>>,
 }
 
-fn main() {
-    let args = CliInputs::parse();
+/// Clone a git repository and cd into it.
+///
+/// Parameters:
+/// url: git clone url
+/// dest: directory where the repo should be cloned
+fn clone_git_repo(url: &str, dest: &str) {
+    env::set_var("GIT_TERMINAL_PROMPT", "0");
+    let git_cmd = Command::new("git")
+        .args(["clone", "--quiet", url, "--depth", "1", dest])
+        .output()
+        .expect("failed to clone repository");
+    // if the git command does not return successfully,
+    // cd command will not work as well. So fail fast.
+    if !git_cmd.status.success() {
+        io::stdout().write_all(&git_cmd.stdout).unwrap();
+        io::stderr().write_all(&git_cmd.stderr).unwrap();
+        return;
+    }
+    println!("Successfully clone repository. Entering repository");
+
+    let dest_path = Path::new(&dest);
+    let _ = env::set_current_dir(&dest_path).is_ok();
     println!(
-        "remote_repo_url: {:?}, feature_branch: {:?},
-        optional_commit_hash: {:?}, optional_rustfmt_config: {:?}",
-        args.remote_repo_url, args.feature_branch, args.commit_hash, args.rustfmt_config
+        "Current directory: {}",
+        env::current_dir().unwrap().display()
     );
 }
+
+fn main() {
+    let _args = CliInputs::parse();
+    let sample_repo = "https://github.com/rust-lang/rustfmt.git";
+    clone_git_repo(sample_repo, "./tmp/");
```

Comment:
What do you think of writing a unit test for `clone_git_repo`? That might be a better more reliable way to ensure that the functionality is working as you intend.